### PR TITLE
Article for Equal Sum Partition

### LINKED ISSUE #146: Article for Equal Sum Partition Problem
## 💥 Proposal

I would like to add an article for Equal Sum Partition Problem. The basic layout for the article will be:

What is Equal Sum Partition Problem?
Explanation for Equal Sum Partition Problem.
Algorithmic Steps
Code (Use C++/Java and Python; At least two languages should be present and pick up code from NeoAlgo with apt credits to author)
Sample I/O
Time & Space Complexity
This article should be between 500-700 words in length. You should use traditional Markdown to prepare the content and the use of external images is prohibited. You can however create your own images. Run a proper Grammarly Lint check before submitting the PR.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

### PR DESCRIPTION
**Pull request by DEEPIKA-SUK**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
Added article for Equal Sum Partition 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #146

### CODE REVIEW

**Comment by DEEPIKA-SUK** [comment]
@HarshCasper Kindly review my PR.

**Comment by vybhav72954** [comment]
Link the issue properly

**Comment by DEEPIKA-SUK** [comment]
> Link the issue properly, Use `#Fixes`

Done

File: docs/dynamic-programming/equal-sum-partition.mdx [hunk 1]
```diff
@@ -0,0 +1,231 @@
+---
+title: Equal Sum Partition
+slug: /dynamic-programming/equal-sum-patition
+---
+
+## Introduction
+
+We have to determine if the given array of positive integers can be divided into two subsets which need not be equal such that the sum of elements of both the subsets is equal. Each element must belong to exactly one subset only.
```
Comment (HarshCasper) [inline docs/dynamic-programming/equal-sum-partition.mdx:8]: Add a bit more description. Right now, it won't make any sense to a reader who might not have came across such a problem.

File: docs/dynamic-programming/equal-sum-partition.mdx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+## Explaination
+
+- We will use the **bottom-up dynamic programming** approach to solve this problem. 
+- Let `n` be the number of elements in the array; `sum` is the sum of all elements of the array. 
+- If the sum is odd, then the answer will be false as we cannot divide the array into two parts having equal sum. 
+- If the sum is even, we will see if we are able to find a subset whose sum is equal to half of `sum`.
```
Comment (HarshCasper) [inline docs/dynamic-programming/equal-sum-partition.mdx:15]: Having a better explanation with diagrams and more would be preferred.

File: docs/dynamic-programming/equal-sum-partition.mdx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+
+## Algorithm
+
+1. Make an array `dp[n+1][sum/2+1]` where row represents array values and column represents sum values from `0` to `sum` (C++ code follows this approach) or `dp[sum/2+1][n+1]` where column represents array values and row represents sum values from `0` to `sum` (Python code follows this approach). We will use first approach for explaining the algorithm.`dp[i][j]` represents that whether we can find the sum `j` from the subset of array elements upto index `i`.
```
Comment (HarshCasper) [inline docs/dynamic-programming/equal-sum-partition.mdx:20]: Break this pointer into sub-pointers

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Why were changes made to `yarn.lock`? Please revert that and fix the merge conflicts.

**Comment by DEEPIKA-SUK** [comment]
> Why were changes made to `yarn.lock`? Please revert that and fix the merge conflicts.

I just did yarn install and then yarn start. Please could you tell me how to remove these changes in yarn lock as I don't know how this file got changed. And please could you also tell me how to resolve this sidebar.js conflict. 
Thank you.

**Comment by DEEPIKA-SUK** [comment]
@HarshCasper I have done the changes, kindly review now.

**Comment by DEEPIKA-SUK** [comment]
I am getting conflicts while pulling the changes so I need to fork the repo again so I am closing this PR and will open the new PR with all the changes done. Sorry for the inconvenience.

**Pull request closed without merging**